### PR TITLE
Full Metrics Web App

### PR DESCRIPTION
**Pull request by Kidd254**
### **Description:**

This pull request is to merge on the Master branch the develop branch of Metricks_Tracker, a mobile web application that allows users to check a list of the population of the world through the use of React and Redux. The application is designed to fetch numeric data from an API related to the population of the world and then present it in an organized and user-friendly manner. The project follows best practices for HTML, CSS, and JavaScript, ensuring a high standard of code quality.
Key Features:
- [ ] Filterable Metrics
    The application's homepage presents a list of categories containing various metrics. Users can easily filter the metrics based on certain parameters. For instance, if users want to see metrics related to specific countries, they can enter the country name in a search field, and the application will filter the metrics accordingly.
- [ ] API Integration:
    The project involves integrating with an external API that provides numeric data based on parameters. This API is open, and the application has been configured accordingly to fetch data securely.
- [ ] Interactive User Interface:
    The user interface of the web app is designed to be engaging and interactive. Numeric values related to each metric category are displayed prominently, making it easy for users to access relevant information quickly. Users can click or tap on a category to navigate to the details page.
- [ ] Details Page for In-Depth Information:
    Upon selecting a specific category from the homepage, users are taken to a details page. Here, the web app fetches additional data from the API to present more in-depth information about the selected category. For example, if the user chooses a specific continent, the details page will display relevant data, such as the number of views for different countries in that continent.
- [ ] Design Guidelines:
    The application adheres to design guidelines that include carefully selected colors, typography, and layout. The overall design concept is attributed to Nelson Sakwa on Behance, and the appropriate credit is given in the README file, in compliance with the Creative Commons license.
- [ ] Testing for Code Reliability:
    To ensure the reliability of the application, unit tests have been implemented for pure functions in the Redux code. Additionally, integration tests using React Testing Library have been created to verify the application's functionality under different scenarios. Mocking techniques have been employed to prevent actual API requests and Redux Store connections during testing.
- [ ] Single-Page Application (SPA) Architecture:
    The web app is built using React and Redux, adhering to the SPA architecture. The data retrieved from the API is stored in the Redux store, which allows for efficient state management and seamless navigation between different pages.
- [ ] Deployment and Online Access:
    The project is successfully deployed and can be accessed online using Render. Users can visit the online version of the app by following the link provided in the README file.
- [ ] Comprehensive Documentation:
    A video presentation has been prepared to showcase the features and functionality of the web app and added to the description of the app on the readme file. The video, limited to 5 minutes, covers a project description, a demonstration of the application's capabilities, and a spotlight on a particularly interesting piece of code or achievement.
I look forward to your feedback and/or approval to proceed with the merge.

Thank you for your time and consideration,
Greetings.
@Kidd254

### CODE REVIEW

File: src/components/Country.jsx [hunk 1]
```diff
@@ -49,7 +49,7 @@ const Country = () => {
             <li className={styles.countryInformationList}>
               {' '}
               Start of Week:
-              {country.startOfWeek}
+              {country.week}
             </li>
           </ul>
         </div>
```
Comment (shella12) [inline src/components/Country.jsx]: Please make sure that your implemented design follows the provided design guidelines. 

- [ ] For your main page please make sure that there is no white space around and your design is responsive. Also the listed countries in the column should have alternative color not the same colors for example if first card is `#5788e5` the second one should be `#4369b2`

Provided design|Your design
---|---
Screenshot 2023-08-05 at 2 47 27 PM|Screenshot 2023-08-05 at 2 47 47 PM

- [ ] Please also make sure that the design of the detailed page also follows the provided guidelines. See the screenshots for comparison 

Provided design|Your design
---|---
Screenshot 2023-08-05 at 2 50 34 PM|Screenshot 2023-08-05 at 2 31 17 PM

**Review by shella12 — CHANGES_REQUESTED** [review]
Hi @Kidd254, 

Good job so far, indeed you have spent time on this project!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
- All tests passed ✅ 
- Organized project structure ✅   
- Correct use of git flow ✅ 
- Used redux for state management ✅ 

## Required Changes ♻️

- [ ] Please make sure to record a 5 minute video demonstrating the features of your website and highlight a small portion of the code you are most proud of.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md [hunk 2]
```diff
@@ -55,7 +55,7 @@ A web-based application to understand about the basics of React - Redux
 
 ## 🚀 Live Demo <a name="live-demo"></a>
 
-- See it live by clicking [Live Demo Link]()
+- See it live by clicking [Live Demo Link](https://metrics-webapp-4iuc.onrender.com/)
```
Comment (algerina) [inline README.md]: - [ ] Well done styling your App however as requested before you still have 2 issues to work on:
-  Record a video presenting the features of the project. with a maximum  length of 5 minutes. The video contains a clear and objective description of the project in professional English. The topic and purpose of the project are made clear. The video contains a demo of the project, highlighting all the project features and a highlight about a piece of the code the student finds interesting or is proud of making.

- [ ] You project is still. not responsive (you still have the white space around for bigger screens):

Screenshot 2023-08-05 at 9 22 50 PM

**Review by algerina — CHANGES_REQUESTED** [review]
Hi @Kidd254 ,

Great job so far!  However you  have some minor issues to work on.



## CHANGES REQUIRED:muscle::clap:

Check the comments under the review.



 
 ## Optional suggestions

 - Please check the comment under the review.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread and tag me @algerina if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)
___
_If anything seems to be unclear to you, reach out and ask anything by commenting on this Pull Request and mention me as [Algerina](https://github.com/algerina), I will receive a notification and will respond._

[GitHub](https://github.com/algerina) | [LinkedIn](https://linkedin.com/in/amel-khiri) | [Twitter](https://twitter.com/AmelQahwadji)

_Reviewed by [Amel Khiri ](https://github.com/algerina)_

**Comment by Kidd254** [comment]
Hello @algerina  we are only doing for mobile version

File: README.md
Comment (aamirkhan2478) [inline README.md:1]: - [ ] If you check your design there are some spacing issues so kindly fix these issues :+1: check below :point_down: the comparison screenshots :slightly_smiling_face: 

**Your Design**
Screenshot from 2023-08-06 03-11-51
Screenshot from 2023-08-06 03-12-11


**Provided Design**
metrics_app

Comment (aamirkhan2478) [inline README.md:1]: - [ ] As mentioned by the previous reviewer :point_up:  highlight a piece of the code that the student finds interesting or is proud of making but in your video, you didn't show the code which finds you proud of making. :slightly_smiling_face:

**Review by aamirkhan2478 — CHANGES_REQUESTED** [review]
Hi @Kidd254, 



Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.

 ## Highlights
✔️ Good Effort
✔️ Your design is good.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by victorokoroji — APPROVED** [review]
Hi @Kidd254, 

# ✔️   Status: APPROVED 🏆 🎉 🎊 

You have done a great job on this project, keep it up 👍 🥇. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
congratulations!  🎉💯🌟 

## Top highlight 🎯

- Requested changes implemented✅ 



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**